curated_transformers/models/attention.py

```diff
@@ -115,6 +116,20 @@ def filter_batch_items(self, mask: Tensor) -> "KeyValueCache":
         return KeyValueCache(key=self.key[mask], value=self.value[mask])
 
 
+class QkvMode(IntEnum):
+    """Modes of handling the query, key and value projections
+    in the self-attention layer.
+
+    ``SEPARATE`` - Use separate projections for query, key and value.
+    ``MERGED_SPLIT_BEFORE`` - Use a merged projection for query, key and value, and split heads before chunking.
+    ``MERGED_SPLIT_AFTER`` - Use a merged projection for query, key and value, and split heads after chunking.
```

Comment:
I took the term _chunking_ from the `chunk` op, but maybe it's not clear what this means? Perhaps something like:

_Use a merged projection for query, key and value, and split heads before splitting the query, key and value representations._

?

Reply:
Done.